[HOTFIX] Resolve trame import issues

banesullivan opened the pull request:
Close #3942, resolves #3941, resolves #3939

Will backport to `release/0.38`

cc @RobPasMue and @germa89

The trame module is opt-in, yet it was added to the top-level imports due to a numpydoc/autosummary issue in #3909.

This is a hotfix to just leave that API undocumented and fix the import issue. We'll have to look into the documentation issue separately.

Further, I added a CI step to make sure the we verify PyVista does not break on import/plot when only installed with it's core dependencies

RobPasMue reviewed (approved):
LGTM @banesullivan thanks for the quick fix!

tkoyama010 commented:
@banesullivan Are we releasing `pyvista==0.38.1` ?

banesullivan commented:
Yes, 0.38.1 will be on its way right after this merge (so give me 30ish minutes)

larsoner reviewed (approved):
Should also fix stuff in `pyvistaqt`, thanks for the quick fix @banesullivan !

larsoner commented:
... also this makes me think somewhere in the CI matrix there should be a "minimal" run where tests are run with minimal dependencies. In MNE we have this and also an "old" CI run with the oldest supported versions. They routinely catch stuff like this that we would have otherwise missed

banesullivan commented:
@larsoner, do you have any insight on if this was a numpydoc error:

```
Extension error (sphinx.ext.autosummary):
Handler <function process_generate_options at 0x7f4eca95ddc0> for event 'builder-inited' threw an exception (exception: no module named pyvista.trame)
make: *** [Makefile:50: html] Error 2
```

(see the diff here: after removing `trame` from the top-level namespace, autosummary couldn't find it)

banesullivan commented:
> ... also this makes me think somewhere in the CI matrix there should be a "minimal" run where tests are run with minimal dependencies. In MNE we have this and also an "old" CI run with the oldest supported versions. They routinely catch stuff like this that we would have otherwise missed

I added a very, very minimal version of this here -- we could do better though. What I added generates a "Report" which actually runs a plot routine under the hood, so it does execute a good chunk of the codebase

larsoner commented:
Where is the error? The doc CI run I see is green

https://github.com/pyvista/pyvista/actions/runs/4075603126/jobs/7022221690

And I don't see any warnings in the log. Did I miss them?

banesullivan commented on doc/api/plotting/trame.rst:
```diff
@@ -2,21 +2,16 @@
-   show_trame
-   elegantly_launch
-   get_or_create_viewer
-   plotter_ui
+.. warning::
+   We are having an issue with Sphinx and numpydoc documenting this module
+   because it is not in the top level namespace of the PyVista package.
+   We will fix this later
```
@larsoner, this is what I was referring to. This breaks when not having `trame` in the top-level namespace. So I just removed it for now to get a patch release out. Having this would cause the error

banesullivan replied:
Sorry to bug you with this... the line between autosummary and numpydoc is blurry to me and thought you might have seen this before

larsoner replied:
No I haven't, and I know `scipy` does not import most of its namespaces (e.g., `scipy.signal`) when you just do `import scipy` so it must be possible to use numpydoc this way. One possibility is that the import fails on the doc build, but it looks like you didn't change the requirements so that doesn't make a lot of sense.

banesullivan replied:
Exactly, this sort of namespacing is super common. Banged my head on the wall over this and wasn't sure what to do so I guess we'll just leave this API undocumented for now

banesullivan replied:
Thanks for taking a look!
